File: Arrays & Hashing/0442. Find All Duplicates in an Array/0442.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        ios_base::sync_with_stdio(false);
        cin.tie(0);
        cout.tie(0);

        vector<int> res;
        for (int& num : nums) {
            int idx = abs(nums) - 1;
            if (nums[idx] < 0) {
                res.push_back(abs(num));
            }
            nums[idx] *= -1;
        }
        return res;
    }
};

File: Arrays & Hashing/0442. Find All Duplicates in an Array/0442.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        # res = []
        # for num in nums:
        #     idx = abs(nums) - 1
        #     if nums[idx] < 0:
        #         res.append(abs(nums))
        #     nums[idx] *= -1
        # return res
    
        res = [0] * (len(nums) + 1)
        for num in nums:
            res[num] += 1

        return [i for i in range(len(res)) if res[i] == 2]            
